app/src/main/java/fr/neamar/kiss/broadcast/InstallShortcutHandler.java

```diff
@@ -0,0 +1,71 @@
+package fr.neamar.kiss.broadcast;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.Intent.ShortcutIconResource;
+import android.graphics.Bitmap;
+import android.util.Log;
+import fr.neamar.kiss.DataHandler;
+import fr.neamar.kiss.KissApplication;
+import fr.neamar.kiss.dataprovider.ShortcutProvider;
+import fr.neamar.kiss.pojo.ShortcutPojo;
+
+public class InstallShortcutHandler extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent data) {
+
+        DataHandler dh = KissApplication.getDataHandler(context);
+        ShortcutProvider sp = dh.getShortcutProvider();
+
+        if (sp == null)
```

Comment:
oh. So turns out you're handling the null case.

Therefore, maybe we should initialize the shortcuthandler only if `prefs.getBoolean("enable-shortcuts", true)`.

Since we're storing image in DB, i think that some app could add shorcuts too fast and we may want to givve the option to the user to "drop" all queries to add shortcuts.

I seem to remember we spoke on this earlier in the opposite sense, saying that we want the shortcuts to be added even if they are not displayed. I'm not so sure on that anymore, what's your opinion?